app/workers/new_page_worker.rb

```diff
@@ -0,0 +1,44 @@
+class NewPageWorker
+  include Sidekiq::Worker
+  attr_reader :page, :doc_html
+
+  def perform(page_id)
+    @page = Page.find(page_id)
+    process
+  rescue SocketError, Gastly::FetchError, OpenURI::HTTPError, Net::OpenTimeout
+    page.update_attributes(status: Page::ERROR_STATUS)
+  end
+
+  private
+
+  def process
+    page.update_attributes(status: Page::IN_PROGRESS_STATUS)
+    @doc_html = download_content
```

Comment:
move this 
@doc_html = download_content
to parse_html_content
looks like you don't this variable here in this method

Reply:
reek didn't like it :)